### PR TITLE
test: DatabaseSelector

### PR DESCRIPTION
**Pull request by yardz**
### SUMMARY
Tests for DatabaseSelector
- Moved to own folder
- Creating tests for the component

### TESTS PLAN
No behavior changes and the interface for importing was maintained.
All tests must pass.

### CODE REVIEW

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    onChange: jest.fn(),
+  };
+});
+
+test('Should render', async () => {
+  const props = { ...defaultProps };
+  await act(async () => {
+    render(<DatabaseSelector {...props} />);
+  });
+  expect(screen.getByTestId('DatabaseSelector')).toBeInTheDocument();
```
Comment (michael-s-molina) [inline superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:166]:
```suggestion
  render(<DatabaseSelector {...props} />);
  expect(await screen.findByTestId('DatabaseSelector')).toBeInTheDocument();
```

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+});
+
+test('Refresh should works', async () => {
+  const props = { ...defaultProps };
+  await act(async () => {
+    render(<DatabaseSelector {...props} />);
+  });
+
+  expect(SupersetClientGet).toBeCalledTimes(2);
```
Comment (michael-s-molina) [inline superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:175]: Maybe use `waitFor` here? 

```
render(<DatabaseSelector {...props} />);

await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(2);
   ...
});

userEvent.click(screen.getByRole('button'));

await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(3);
   ...
});


```

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+  const selector = screen.getByText('Schema:');
+  expect(selector).toBeInTheDocument();
+  expect(selector.parentElement?.textContent).toBe('Schema: public');
+
+  await act(async () => {
+    userEvent.click(screen.getByRole('button'));
+  });
+
+  expect(screen.getByText('Select a schema (2)')).toBeInTheDocument();
```
Comment (michael-s-molina) [inline superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:224]:
```suggestion
  render(<DatabaseSelector {...props} />);

  const selector = await screen.findByText('Schema:');
  expect(selector).toBeInTheDocument();
  expect(selector.parentElement).toHaveTextContent('Schema: public');

  userEvent.click(screen.getByRole('button'));

  expect(await screen.findByText('Select a schema (2)')).toBeInTheDocument();
```

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+test('Should database select display options', async () => {
+  const props = { ...defaultProps };
+  await act(async () => {
+    render(<DatabaseSelector {...props} />);
+  });
+
+  const selector = screen.getByText('Database:').parentElement;
+
+  expect(selector).toBeInTheDocument();
+  expect(selector?.textContent).toBe('Database:postgresql examples');
```
Comment (michael-s-molina) [inline superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:207]:
```suggestion
  render(<DatabaseSelector {...props} />);
  const selector = await screen.findByText('Database:');
  expect(selector).toBeInTheDocument();
  expect(selector.parentElement).toHaveTextContent(
    'Database:postgresql examples',
  );
```

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+import React from 'react';
+import { render, act, screen } from 'spec/helpers/testing-library';
+import { SupersetClient } from '@superset-ui/core';
+import userEvent from '@testing-library/user-event';
+import DatabaseSelector from '.';
+
+const SupersetClientGet = jest.spyOn(SupersetClient, 'get');
+
+let defaultProps: any = {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:28]: `defaultProps` is being overridden by `beforeEach` and has `any` type. If the tests don't modify it, we can turn it into a `const`.

Comment (yardz) [reply superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:28]: done: https://github.com/apache/superset/pull/13753#discussion_r600862661

File: superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx (hunk 2 above)
Comment (yardz) [reply superset-frontend/src/components/DatabaseSelector/DatabaseSelector.test.tsx:173]: In this case the results are different. I want to guarantee the exact number of calls.
```
await waitFor(() => {
   expect(SupersetClientGet).toBeCalledTimes(2); // this is valid even when calling 2 or more times
});
```
```
expect(SupersetClientGet).toBeCalledTimes(2); // this is valid even when calling only 2 times
```

The intention here is to ensure that calls are being made exactly this number of times.

**Review by michael-s-molina — APPROVED** [review]
LGTM. 

If you want, we can move `SupersetClientGet.mockImplementation` outside of `beforeEach` because it seems that it only needs to be run once.

**Review by rusackas — APPROVED** [review]
LGTM

**Pull request merged**